compiler/plc_ast/src/ast.rs

```diff
@@ -1156,6 +1017,56 @@ impl AstStatement {
     pub fn is_literal(&self) -> bool {
         matches!(self, AstStatement::Literal { .. })
     }
+    pub fn set_location(self, new_location: SourceLocation) -> Self {
```

Comment:
this looks pretty strange.
it looks like a normal setter - although it returns a completely new Self,

... I changed this in the Node-commit

Reply:
I agree, should I rename or do we handle this in your commit?